hack/dind-cluster.sh

```diff
@@ -44,59 +43,58 @@
+#    OPENSHIFT_CONFIG_ROOT=[path] hack/dind-cluster.sh [command]
 #
-# Running Tests
-# -------------
-#
-# This script includes a shortcut for running the networking e2e
-# tests.  The test-net-e2e command will run the extended networking
-# tests against an already running dind cluster:
-#
-#     hack/dind-cluster.sh test-net-e2e
-#
-# Bash Aliases
-# ------------
-#
-# The following bash aliases are available in the cluster containers:
+# Suggested Workflow
+# ------------------
 #
-# oc-create-hello - create the 'hello' example app
-# oc-less-log - invoke 'less' on the openshift daemon log (will target
-#               the master or node log depending on the type of node)
-# oc-tail-log - invoke tail on the openshift daemon log
+# When making changes to the deployment of a dind cluster or making
+# breaking golang changes, the 'restart' command will ensure that an
+# existing cluster is cleaned up before deploying a new cluster.
 #
-# Process Management
-# ------------------
+# When only making non-breaking changes to golang code, the 'redeploy'
+# command avoids restarting the cluster.  'redeploy' rebuilds the
+# openshift binaries and deploys them to the existing cluster.
 #
-# Due to docker-in-docker conflicting with systemd when running in a
-# container, supervisord is used instead.  The 'supervisorctl' command
-# is the equivalent of 'systemctl' and logs for managed processes can
-# be found in /var/log/supervisor.
+# Running Tests
+# -------------
 #
-# Loopback Devices
-# ----------------
+# The extended tests can be run against a dind cluster as follows:
 #
-# Due to the way docker-in-docker daemons interact with loopback
-# devices, it is important to invoke 'dind-cluster.sh stop' on a
-# running cluster instead of manually stopping the containers.  This
-# ensures that the containerized docker daemons are gracefully
-# shutdown and allowed to release their loopback devices before
-# container shutdown.  If the daemons are not stopped before container
-# shutdown, the associated loopback devices will be effectively
-# unusable ('leaked') until a subsequent host reboot.  If enough
-# loopback devices are leaked, cluster boot may not be possible since
-# each openshift node running in a container depends on a docker
-# daemon requiring 2 loopback devices.
+#     OPENSHIFT_CONFIG_ROOT=dind test/extended/networking.sh
 
 set -o errexit
 set -o nounset
 set -o pipefail
 
-source $(dirname "${BASH_SOURCE}")/dind/init.sh
+DIND_MANAGEMENT_SCRIPT=true
+
+source $(dirname "${BASH_SOURCE}")/../contrib/vagrant/provision-config.sh
+
+# Enable xtrace for container script invocations if it is enabled
+# for this script.
+BASH_CMD=
+if [ "$(set | grep xtrace)" ]; then
+    BASH_CMD="bash -x"
+fi
+
+DOCKER_CMD=${DOCKER_CMD:-"sudo docker"}
+
+# Override the default CONFIG_ROOT path with one that is
+# cluster-specific.
+CONFIG_ROOT=${OPENSHIFT_CONFIG_ROOT:-/tmp/openshift-dind-cluster/${INSTANCE_PREFIX}}
+
+DEPLOY_SSH=${OPENSHIFT_DEPLOY_SSH:-false}
```

Comment:
Is there any reason to not just always deploy ssh in the dind cluster? It doesn't seem like it would take much time/space/memory, and it means you're always able to use the debug script. (Though, alternatively, we could just make the debug script be dind-cluster-aware and use 'docker exec' rather than 'ssh' in that case.)